### PR TITLE
new feature dynamic IP's list

### PR DESCRIPTION
**Pull request by Omer-Sadeh**
replaced the dropdown menu of IPs with a pop up window that opens when pressing the button, which inside we have a dynamic list, and an option to add new IPs and delete existing ones.
Original dropdown menu:
old dropdown menu
old ui
New Dynamic List:
new dynamic IP list
the button:
new ui 1
the new pop up window:
new ui 2
and the delete button, after pressing it:
new ui 3
In order to do that, I modified the states of App, changing the name of 2 of them to fit better with current code, and adding 2 (1 toggle of the pop up window, Modal, and 1 which is an parallel array to the IP's array, for the Delete button I added):
states edit
Also, I made new functions, for the toggles and switching:
new func 1
and for the buttons themselves and the deletion operation:
new func 2

### CODE REVIEW

**Comment by lomudi** [comment]
@Omer-Sadeh 
1. please notice that the CI deploy tests where failed, please fix it.
2. about the PR description - it is great that you are providing all the details, in general code changes can be easily seen under the "file changed" tab, so no need to put screenshots of code changes... in the other hand, screenshots of the UI changes are very valuable for the reviewer - please add those to the PR description. 

Thx! I have asked @EtayTod to review the PR.

**Comment by Omer-Sadeh** [comment]
fixed the CI deploy issues, it required removing un-used imports and a small fix in RobotObject.tsx.
also, added more screenshots to description.

File: src/App.tsx [hunk 1]
```diff
@@ -2,21 +2,21 @@ import React from 'react';
 import './App.scss';
 import { HttpClient } from '@butter-robotics/mas-javascript-api';
 import { RobotObject } from './components/RobotObject';
-import { useState } from 'react';
```
Comment (Todress) [inline src/App.tsx:5]: why removing this import?

File: src/App.tsx [hunk 2]
```diff
@@ -60,9 +56,65 @@ export class App extends React.PureComponent<{}, AppState> {
...
+
+	renderButtons(ip: string) {
+		let IParray = [...this.state.labCurrentIPs];
+		let StateArray = [...this.state.IPdeleteState];
+		let index = IParray.indexOf(ip);
+		if (this.state.IPdeleteState[index]) {
+			StateArray[index] = false;
+			return(
+				<>
+				<Button variant="secondary" disabled>Are you sure you want to delete {ip}?</Button>
```
Comment (Todress) [inline src/App.tsx]: look at the css of other buttons, we should follow one line of design here

File: src/components/RobotObject.tsx [hunk 3]
```diff
@@ -9,7 +9,7 @@ export function RobotObject({ butterClient, onRemove }: { butterClient: HttpClie
 
   useEffect(() => {
     loadAnimations();
-  }, []);
```
Comment (Todress) [inline src/components/RobotObject.tsx]: why removing []?

File: src/components/ScenariosButtons.tsx [hunk 4]
```diff
@@ -1,16 +1,13 @@
...
-import { start } from 'repl';
 
 const kip1 = new HttpClient('192.168.57.30');
 const kip2 = new HttpClient('192.168.56.188');
 
 kip1.timeout = 240;
 kip2.timeout = 240;
 
-let interval1: NodeJS.Timeout;
-let interval2: NodeJS.Timeout;
```
Comment (Todress) [inline src/components/ScenariosButtons.tsx:13]: aren't we using those intervals? did you perform a live test on few robots to verify all functioning well?

File: src/index.tsx [hunk 5]
```diff
@@ -3,7 +3,6 @@ import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App';
 import * as serviceWorker from './serviceWorker';
-import { BrowserRouter, Route, Switch } from "react-router-dom"
```
Comment (Todress) [inline src/index.tsx]: same - aren't we using this?

File: src/App.tsx [hunk 6]
```diff
@@ -25,10 +25,12 @@ export class App extends React.PureComponent<{}, AppState> {
...
 	state: AppState = {
 		dayNightStatus: false,
-		currentIPInput: '192.168.57.30',
+		NewIPInput: '',
 		currentButterClients: [],
-		show: false,
+		showInst: false,
+		showNewIP: false,
 		labCurrentIPs: ['192.168.57.30', '192.168.57.32', '192.168.57.34', '192.168.56.188', '192.168.56.193', '192.168.56.206'],
+		IPdeleteState: [false, false, false, false, false, false],
```
Comment (Todress) [inline src/App.tsx]: what happens to this list after you delete and then add new ip?

**Review by Todress — CHANGES_REQUESTED** [review]
please note my comments, if something is not clear - reply here to the comments.
beside the code review, did you perform an actual test with the robots?
beside that - looking great! 👍

File: src/App.tsx (hunk 6 above)
Comment (Todress) [reply src/App.tsx]: make sure that you don't mix states with different ip's after deleting and adding some

File: src/App.tsx (hunk 1 above)
Comment (Omer-Sadeh) [reply src/App.tsx:5]: The PR deploy test failed when I had warnings in the code, such as "defined but never used", so i have to delete this never used line.

File: src/components/RobotObject.tsx (hunk 3 above)
Comment (Omer-Sadeh) [reply src/components/RobotObject.tsx]: The PR deploy test failed when I had warnings in the code, and this line had a warning for unnecessary [].

File: src/components/ScenariosButtons.tsx (hunk 4 above)
Comment (Omer-Sadeh) [reply src/components/ScenariosButtons.tsx:13]: The PR deploy test failed when I had warnings in the code, such as "defined but never used", so i have to delete this never used line. 
I didn't check it in the lab, but we can keep this lines, as I said, only removed them to get rid of the warnings to pass the deploy tests.

File: src/index.tsx (hunk 5 above)
Comment (Omer-Sadeh) [reply src/index.tsx]: The PR deploy test failed when I had warnings in the code, such as "defined but never used", so i have to delete this never used line.

File: src/App.tsx (hunk 2 above)
Comment (Omer-Sadeh) [reply src/App.tsx]: would love some more explainations on this.

File: src/App.tsx (hunk 6 above)
Comment (Omer-Sadeh) [reply src/App.tsx]: you can see in the code, where you add and remove IPs, it updates the new array as well, and I tested it and it works.
It checks the index of the IP in the IPs array, and when removing it, it removes the corresponding index in the states array.

**Comment by Omer-Sadeh** [comment]
Added back all the lines I deleted that caused warnings, and changed the syntax in a line in RobotObject component, to be according to the same line in Develop, as seen in the picture:
Capture

**Review by Todress — APPROVED** [review]
Approved

**Pull request merged**